sympy/ntheory/ecm.py

```diff
@@ -0,0 +1,244 @@
+from sympy.ntheory import sieve, isprime
+from sympy.core.power import integer_log
+from sympy.core.compatibility import as_int
+import random
+
+#----------------------------------------------------------------------------#
+#                                                                            #
+#                   Lenstra's Elliptic Curve Factorization                   #
+#                                                                            #
+#----------------------------------------------------------------------------#
+
+
+class Point:
+    """Montgomery form of Points in an elliptic curve.
+    In this form, the addition and doubling of points
+    does not need any y-coordinate information thus
+    decreasing the number of operation.
+
+    Parameters:
+    ===========
+
+    x_corr : X coordinate of the Point
+    z_corr : Z coordinate of the Point
+    a_24 : Parameter of the elliptic curve in Montgomery form
+    mod : modulus
+
+    References
+    ==========
+
+    .. [1]  http://www.hyperelliptic.org/tanja/SHARCS/talks06/Gaj.pdf
+    """
+
+    def __init__(self, x_corr, z_corr, a_24, mod):
+        self.x_corr = x_corr
+        self.z_corr = z_corr
+        self.a_24 = a_24
+        self.mod = mod
+
+    def add(self, Q, diff):
+        """
+        Add two points P and Q where diff = P - Q.
+
+        Parameters:
+        ===========
+
+        Q : point on the curve in Montgomery form
+        diff : P - Q
+        """
+        u = (self.x_corr - self.z_corr)*(Q.x_corr + Q.z_corr)
+        v = (self.x_corr + self.z_corr)*(Q.x_corr - Q.z_corr)
+        add, subt = u + v, u - v
+        x_corr = diff.z_corr*add*add % self.mod
+        z_corr = diff.x_corr*subt*subt % self.mod
+        return Point(x_corr, z_corr, self.a_24, self.mod)
+
+    def double(self):
+        """
+        Doubles a point in an elliptic curve in Montgomery form.
+        """
+        u, v = self.x_corr + self.z_corr, self.x_corr - self.z_corr
+        u, v = u*u, v*v
+        diff = u - v
+        x_corr = u*v % self.mod
+        z_corr = diff*(v + self.a_24*diff) % self.mod
+        return Point(x_corr, z_corr, self.a_24, self.mod)
+
+    def mont_ladder(self, k):
+        """
+        Scalar multiplication of a point in Montgomery form
+        using Montgomery Ladder Algorithm.
+
+        Parameters:
+        ==========
+
+        k : The positive integer multiplier
+        """
+        Q = self
+        R = self.double()
+        for i in bin(k)[3:]:
+            if  i  == '1':
+                Q = R.add(Q, self)
+                R = R.double()
+            else:
+                R = Q.add(R, self)
+                Q = Q.double()
+        return Q
+
+
+def ecm_one_factor(n, B1=10000, B2=100000, max_curve=200, seed=1234):
+    """Returns one factor of n using
+    Lenstra's 2 stage Elliptic curve Factorization
+    with Suyama's Parameterization. Here Montgomery
+    arthematic is used for fast computation of addition
```

Comment:
`arthematic`->`arithematic`

Reply:
Neither of those spellings is correct :)

Reply:
corrected.